packages/network/src/logic/node/ProxyStreamConnectionManager.ts

```diff
@@ -0,0 +1,211 @@
+import { TrackerManager } from './TrackerManager'
+import { StreamManager } from './StreamManager'
+import { NodeToNode } from '../../protocol/NodeToNode'
+import { Event, Node, NodeId } from './Node'
+import {
+    PublishStreamConnectionRequest,
+    PublishStreamConnectionResponse,
+    SPID,
+    SPIDKey,
+    UnsubscribeRequest
+} from 'streamr-client-protocol'
+import { promiseTimeout } from '../../helpers/PromiseTools'
+import { Logger } from '../../helpers/logger/LoggerNode'
+const logger = new Logger(module)
+
+export interface ProxyStreamConnectionManagerOptions {
+    trackerManager: TrackerManager,
+    streamManager: StreamManager,
+    nodeToNode: NodeToNode,
+    node: Node,
+    nodeConnectTimeout: number,
+    acceptProxyConnections: boolean
+}
+
+enum State {
+    CONNECTING,
+    CONNECTED,
+    RECONNECTING
+}
```

Comment:
I would be a bit careful naming the states like this. As we've previously noticed, we don't really know if a connection is connected or not until we attempt to use it. Maybe domain-level terms would be better e.g. `NEGOTIATING`, `ACCEPTED` . Also generally `CONNECTING` and `RECONNECTING` can be just collapsed to one state.

Also I don't see that `state` would be used much here, just for bookkeeping? We mostly write the value but don't read it anywhere?

Reply:
At the moment, the only state that is really used is the reconnecting state